### PR TITLE
bugfix: Added base key validity period of 60 minutes.

### LINKED ISSUE #764: Headscale fails to activate clients with postgresql backend
**Bug description**

Tailscale clients authenticate successfully with headscale when headscale is configured to use postgres but then get stuck in a loop and keep refreshing keys. 

More specifically, 

In case of sqlite, 
1. Clients sends 556byte payload, headscale answers it. 
2. Then, Client sends ~629 byte payload containing the peerapi4, peerapi6 services and client responds 
3. Then, client sends a 1000+ byte payload with read_only=false containing it's endpoints and after this everything works!

In case of postgres, 
1. Client sends 556byte payload, headscale returns the exact same answer as sqlite. 
2. Client sends 629byte payload(read only=true), headscale responds but KeyExpiry is set to `0001-01-01 05:53:28+05:53:28`. 
4. Client again sends a 629byte payload and this keeps happening in a loop. 
Client never sends a payload with read_only=false. 

**To Reproduce**

1. Install postgresql 14
2. Install latest build of headscale(from git head or the latest release) 
3. Configure it to use postgresql and run it

Try to register any tailscale client.

**Context info**

### PR DESCRIPTION
**Pull request by ishanjain28**
- [x] read the [CONTRIBUTING guidelines](README.md#contributing)
- [x] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [x] updated documentation if needed
- [x] updated CHANGELOG.md

Fixes #764 

PR Description,

Currently headscale initializes machine key's expiry time with `&time.Time{}`. This is formatted to, `0001-01-01T00:00:00Z`(in RFC3339 format). 

This means, The keys expiry time will always be in the past and clients will be stuck in a loop renewing the key. 

This PR addresses this problem by adding a minimum validity of 60 minutes to the generated keys.

### CODE REVIEW

**Comment by ishanjain28** [comment]
This issue has not been fixed completely. 

Android clients are working fine and are getting the correct expiry time but Linux clients are still getting, 

> expiry                 | 0001-01-01 05:53:28+05:53:28

**Comment by ishanjain28** [comment]
Fixed.

1. OIDC renewal code mentios that the function accepts current time but then passes, `&time.Time{}`. This has been changed to pass, `time.Now().Add(DefaultKeyExpireTime)`. 
2. In protocol_common.go, In the refresh operation it was not renewing the key. `expire` field in database was being overwritten to `0001-01-01T00:00:00Z`.

File: protocol_common.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ const (
 	//
 	// See also https://github.com/tailscale/tailscale/blob/main/tailcfg/tailcfg.go
 	NoiseCapabilityVersion = 39
+
+	// DefaultKeyExpireTime determines the valid period of a generated key
```
Comment (juanfont) [inline protocol_common.go:30]: Comment should end in a period

File: protocol_common.go [hunk 2]
```diff
@@ -26,6 +26,10 @@ const (
...
+	// Keys of new machines are valid for this period
```
Comment (juanfont) [inline protocol_common.go:31]: Comment should end in a period

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 0.17.1 (2022-XX-XX)
```
Comment (juanfont) [inline CHANGELOG.md:3]:
```suggestion
```

Comment (juanfont) [reply CHANGELOG.md:3]: Can you please add this under 0.17.0 ?

File: protocol_common.go [hunk 4]
```diff
@@ -26,6 +26,10 @@ const (
...
+	DefaultKeyExpireTime = 60 * time.Minute
```
Comment (juanfont) [inline protocol_common.go:32]: Any rationale behind 60 minutes?

**Comment by juanfont** [comment]
Digging into this, upstream only seems to say about key expiration this (in `RegisterRequest`)

	// Expiry optionally specifies the requested key expiry.
	// The server policy may override.
	// As a special case, if Expiry is in the past and NodeKey is
	// the node's current key, the key is expired.


Perhaps could be DefaultKeyExpireTime could be something configurable, with a default time of 60 minutes...

**Comment by kradalby** [comment]
Hi! as part of #1473, we have reorganised a lot of the code. 

To clear PRs that needs to be rebased or redone, we are closing open PRs that will require significant code change to be merged.

In addition, the issue of the PR might in some cases have been fixed, change or no longer relevant, so it would be great if this is considered as well.

Thank you for your contribution! 

If it is still relevant and the PR is reopened, we will aim at getting the changes into the next release after the reorg if accepted.

**Pull request closed without merging**